src/mongo_logger.ts

```diff
@@ -415,12 +415,27 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
-  if (typeof value === 'function') {
+  if (typeof value === 'string') {
+    strToTruncate = value;
+  } else if (typeof value === 'function') {
     strToTruncate = value.toString();
   } else {
     strToTruncate = EJSON.stringify(value, options);
   }
 
+  // handle truncation that occurs in the middle of multi-byte codepoints
+  if (
+    maxDocumentLength !== 0 &&
+    strToTruncate.length > maxDocumentLength &&
+    strToTruncate.charCodeAt(maxDocumentLength - 1) !==
+      strToTruncate.codePointAt(maxDocumentLength - 1)
```

Comment:
I'm not the most well versed on Unicode, so a general question: Are there valid multi-byte codepoints where these two functions return the same value? I'd imagine there aren't but I just want to confirm.

Reply:
There shouldn't be. 
Here, we assume 1 unicode codepoint = 2 unicode code units. Valid multibyte codepoints may have `codePointAt(codeUnitIndex) === charCodeAt(codeUnitIndex) ` can be true for the second code unit in the second but not the first. `charCodeAt` only returns values as high as until 65535 (the number of code units), and for multibyte characters, the code point value must be higher than 65535 for the first code unit. 

There is more context [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt).